includes/discovery/sensors/dbm/ironware.inc.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * LibreNMS Interface Power dBm module for Brocade IronWare
+ *
+ * © 2017 Chris A. Evans <thecityofguanyu@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+echo 'Brocade IronWare Interface dBm ';
+
+$multiplier = 1;
+$divisor = 1000;
+
+foreach ($pre_cache['ironware_optic_oids'] as $index => $entry) {
+    if ($entry['snIfOpticalMonitoringRxPower']) {
+        if (strpos($entry['snIfOpticalMonitoringRxPower'], 'N\/A') !== true) {
+            $oid = '.1.3.6.1.4.1.1991.1.1.3.3.6.1.3.' . $index;
+            $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifIndex`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Rx Power';
+            $currentrx = $entry['snIfOpticalMonitoringRxPower'];
```

Comment:
That's a lot of _replaces, what's the reason?

Reply:
To strip off unnecessary data from the string returned in the snmpwalk. The devices send optical you alarm state, "dBm, plenty of whitespace, and wraps it all in double quotes.

Example output:

```
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.1 = STRING: " -006.5580 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.2 = STRING: " -214748.3647 dBm: Low-Alarm "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.3 = STRING: " -006.5777 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.4 = STRING: "  000.6625 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.5 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.6 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.7 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.8 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.9 = STRING: "     N/A     "
```

If you have a recommendation to consolidate those various replace functions into less statements, then let me know.

Reply:
Maybe we should put something like this:

https://github.com/librenms/librenms/pull/7384/files#diff-67aaf7dd786395ce4cbd2c3996cf1bdcR1101

In a function `return_num()`?

Reply:
This definitely isn't the first time I've needed to rip out non-numeric characters, so an added function for that sounds like a great idea.

Reply:
I can start writing something up and add it to my PR, no big deal. Just correct/improve it as you see fit.

Reply:
👍